runtime/compilers/rillv1/parse_api.go

```diff
@@ -1,14 +1,64 @@
 package rillv1
 
 import (
+	"encoding/json"
 	"fmt"
 
+	"github.com/go-openapi/spec"
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
 // APIYAML is the raw structure of a API resource defined in YAML (does not include common fields)
 type APIYAML struct {
 	DataYAML `yaml:",inline" mapstructure:",squash"`
+	OpenAPI  *OpenAPIYAML `yaml:"openapi"`
+}
+
+type OpenAPIYAML struct {
+	Info     map[string]any `yaml:"info"`
+	Request  *ReqSpecYAML   `yaml:"request"`
+	Response *RespSpecYAML  `yaml:"response"`
+}
+
+type ReqSpecYAML struct {
+	Summary    string           `yaml:"summary"`
+	Parameters []map[string]any `yaml:"parameters"`
+}
+
+type RespSpecYAML struct {
+	Description string         `yaml:"description"`
+	Schema      map[string]any `yaml:"schema"`
+}
```

Comment:
I would consider flattening or reworking this slightly – a few comments:
1. The `summary` is not related to the request itself, but to the endpoint as a whole
2. We might consider omitting the response `description`. Unlike the full OpenAPI spec, the user here can only define a single response schema, so we should be able to autopopulate the description with something like "Successful resolver response".
3. The `ReqSpecYAML` and `RespSpecYAML` type names should maybe be prefixed by `OpenAPI` since the package is quite large, to avoid polluting the scope. Or maybe just be inline structs (see `AlertYAML` for an example)?